src/ui/controls/color-picker/hsb-picker.tsx

```diff
@@ -118,15 +130,22 @@ export default function HSBPicker(props: HSBPickerProps) {
     }
 
     function onSBCanvasRender(canvas: HTMLCanvasElement) {
+        if (isElementHidden(canvas)) {
+            return;
+        }
         const hue = activeHSB.h;
         const prevHue = prevColor && rgbToHSB(parse(prevColor)).h;
-        if (hue === prevHue) {
-            return;
+        if (store.wasPrevHidden || hue !== prevHue) {
+            store.wasPrevHidden = false;
+            renderSB(hue, canvas);
         }
-        renderSB(hue, canvas);
     }
 
-    function onHueCanvasCreate(canvas: HTMLCanvasElement) {
+    function onHueCanvasRender(canvas: HTMLCanvasElement) {
+        if (isElementHidden(canvas) || store.hueCanvasRendered) {
```

Comment:
Replacing vice-versa should be more efficient (checking for the flag before calling the function), a micro optimisation.